src/effects/RealtimeEffectState.cpp

```diff
@@ -213,20 +213,16 @@ struct RealtimeEffectState::Access final : EffectSettingsAccess {
    std::weak_ptr<RealtimeEffectState> mwState;
 };
 
-RealtimeEffectState::RealtimeEffectState(const PluginID & id)
+RealtimeEffectState::RealtimeEffectState(const PluginID& id)
 {
    SetID(id);
+   BuildAll();
```

Comment:
I think BuildAll is now unnecessary for correct lifetimes of `RealtimeEffectStateUI`, now that the state objects are shared across Undo history states.

Undo/Redo still closes the window.  If there is an enhancement to avoid that, remember that test case should be:
- Add an effect to a stack but don't open settings yet
- make some undoable edit
- undo
- click the gear
- redo (here, the sidebar must reconnect to another track)
- click the gear again
And the second click should hide the dialog, not open a second dialog.

Reply:
Let's keep it for now, probably, it does no harm either

Reply:
If any more ClientData are attached to the RealtimeEffectState, they too would be built non-lazily.

Well we don't care about that now.